cognite_toolkit/_cdf_tk/client/api/lookup.py

```diff
@@ -276,16 +277,28 @@ def _read_acl(self) -> Capability:
 
 class LocationFiltersLookUpAPI(AllLookUpAPI):
     def _lookup(self) -> None:
+        def _recursive_lookup(locs: LocationFilterList) -> None:
+            for loc in locs:
+                if loc.external_id and loc.id:
+                    self._cache[loc.external_id] = loc.id
+                    self._reverse_cache[loc.id] = loc.external_id
+                if loc.locations:
+                    _recursive_lookup(loc.locations)
+
         location_filters = self._toolkit_client.location_filters.list()
-        self._cache = {location_filter.external_id: location_filter.id for location_filter in location_filters}
-        self._reverse_cache = {location_filter.id: location_filter.external_id for location_filter in location_filters}
+        _recursive_lookup(location_filters)
 
     def _read_acl(self) -> Capability:
         return LocationFiltersAcl(
             [LocationFiltersAcl.Action.Read],
             scope=LocationFiltersAcl.Scope.All(),
         )
 
+    def _id(self, external_id: SequenceNotStr[str]) -> dict[str, int]:
+        if not self._has_looked_up:
+            self._lookup()
+        return {external_id: self._cache[external_id] for external_id in external_id if external_id in self._cache}
```

Comment:
Same issue here, I think we need to have a discussion on how to handle the case when an `externalID` is missing in CDF. I think it should give a warning to the user and not pass by silently.